ReactiveCocoa/UIKit/iOS/UIKeyboard.swift

```diff
@@ -37,19 +68,31 @@ public struct KeyboardChangeContext {
 		return base[UIKeyboardIsLocalUserInfoKey] as! Bool
 	}
 
-	fileprivate init(_ userInfo: [AnyHashable: Any]) {
+	fileprivate init(userInfo: [AnyHashable: Any], event: KeyboardEvent) {
 		base = userInfo
+		self.event = event
 	}
 }
 
 extension Reactive where Base: NotificationCenter {
+	/// Create a `Signal` that notifies whenever the system keyboard announces specified events.
+	///
+	/// - returns: A `Signal` that emits the context of system keyboard events.
+	public func keyboard(_ events: KeyboardEvent...) -> Signal<KeyboardChangeContext, NoError> {
+		return .merge(
+			events.map { event in
+				notifications(forName: event.notificationName)
+					.map { notification in KeyboardChangeContext(userInfo: notification.userInfo!, event: event) }
+			}
+		)
+	}
+	
 	/// Create a `Signal` that notifies whenever the system keyboard announces an
 	/// upcoming change in its frame.
 	///
 	/// - returns: A `Signal` that emits the context of every change in the
 	///            system keyboard's frame.
 	public var keyboardChange: Signal<KeyboardChangeContext, NoError> {
-		return notifications(forName: .UIKeyboardWillChangeFrame)
-			.map { notification in KeyboardChangeContext(notification.userInfo!) }
```

Comment:
I think this should continue to use the old code just because it'd likely be more performant.